app/javascript/mastodon/db/modifier.js

```diff
@@ -0,0 +1,86 @@
+import asyncDB from './async';
+
+const limit = 1024;
+
+function put(name, objects, callback) {
+  asyncDB.then(db => {
+    const store = db.transaction(name, 'readwrite').objectStore(name);
+    const index = store.index('id');
+    let left = objects.length;
+
+    function done() {
+      left--;
+
+      if (left <= 0) {
+        store.count().onsuccess = count => {
+          const excess = count.target.result - limit;
+
+          if (excess > 0) {
+            store.getAll(null, excess).onsuccess =
```

Comment:
Edge doesn't support `getAll`/`getAllKeys` yet. (Sorry 😅) There is a polyfill you can use: https://github.com/dumbmatter/IndexedDB-getAll-shim

Reply:
Also in this case it appears you can use `getAllKeys` rather than `getAll` because you are just returning the `id`s, right?

Reply:
The key is not `id`.